modules/dnn/src/layers/cpu_kernels/conv_block.simd.hpp

```diff
@@ -494,10 +494,12 @@ void convBlockMR1_F32(int np, const float * a, const float * b, float *c, const
 void convBlock_F16(int np, const char * _a, const char * _b, char * _c, int ldc, bool init_c, int width,
                     const int convMR_fp16, const int convNR_fp16)
 {
+#if !defined(OPENCV_HIDE_FLOAT16_T)
     typedef __fp16 float16_t;
-    const float16_t* a = (const float16_t*)_a;
-    const float16_t* b = (const float16_t*)_b;
-    float16_t* c = (float16_t*)_c;
+#endif
```

Comment:
Why do we still need this `float16_t`?

Reply:
Because `vdupq_n_f16` at line 505 is an macro defined in Clang `arm_neon.h` (details in https://github.com/opencv/opencv/issues/25210#issue-2183606808). In this macro, `float16_t` is used, so this typedef has to be saved to make `vdupq_n_f16` works unless we undo `typedef hfloat float16_t` in this file.

Note that `vdupq_n_f16` in GCC `arm_neon.h` is an function.

Reply:
They all are dropped now.